src/main/resources/templates/feed.html

```diff
@@ -29,8 +29,16 @@ <h1 th:text="${entries.size()} + ' entries'"></h1>
         </tbody>
     </table>
 
-    <a th:if="${previousPageLink}" th:href="${previousPageLink}">previousPage</a>
-    <a th:if="${nextPageLink}" th:href="${nextPageLink}">nextPage</a>
+    <li th:if="${pagination.hasPreviousPage()}" class="previous">
```

Comment:
`<li>` isn't valid here, because you're not in a list.  What are you trying to do?